src/serena/util/file_system.py

```diff
@@ -138,21 +137,39 @@ def __init__(self, repo_root: str) -> None:
 
     def _load_gitignore_files(self) -> None:
         """Load all gitignore files from the repository."""
-        gitignore_files = self._find_gitignore_files()
-
-        for gitignore_file in gitignore_files:
-            spec = self._create_ignore_spec(gitignore_file)
+        for gitignore_path in self._iter_gitignore_files():
+            log.info("Processing gitignore file: %s", gitignore_path)
+            spec = self._create_ignore_spec(gitignore_path)
             if spec.patterns:  # Only add non-empty specs
                 self.ignore_specs.append(spec)
 
-    def _find_gitignore_files(self) -> list[str]:
+    @dataclass
+    class _QueueItem:
+        abs_path: str
+        rel_path: str | None = None

```

Comment:
None is only ever passed for the project root, right? Instead of a default value, we could explicitly pass empty string and have the type being string.

I don't fully understand why we need the absolute path at all, seems like duplicated information if we have the repo root available

Reply:
It seems overengineered, I don't see why not just a list of relative paths. Introducing a data structure just for the root to not be listed as "" seems like an overkill. It also allows incorrect state in principle